ca/server.go

```diff
@@ -579,30 +582,49 @@ func (s *Server) UpdateRootCA(ctx context.Context, cluster *api.Cluster) {
 		if err := s.securityConfig.UpdateRootCA(&updatedRootCA, externalCARootPool); err != nil {
 			logger.WithError(err).Error("updating Root CA failed")
 		} else {
-			logger.Debugf("Root CA updated successfully")
+			// only update the server cache if we've successfully updated the root CA
+			logger.Debug("Root CA updated successfully")
+			s.rootCA = &cluster.RootCA
```

Comment:
Let's make this `cluster.RootCA.Copy()` to avoid storing a pointer into the object living in the store. Just in case we were to write into it for some reason...